src/julia.h

```diff
@@ -188,7 +188,8 @@ typedef struct _jl_sym_t {
     struct _jl_sym_t *left;
     struct _jl_sym_t *right;
     uptrint_t hash;    // precomputed hash value
-    JL_ATTRIBUTE_ALIGN_PTRSIZE(char name[]);
+    // variable length
+    JL_ATTRIBUTE_ALIGN_PTRSIZE(char name[1]);
```

Comment:
for these, maybe we should just eliminate these convenience members and use macros instead to point to these elements as `#define jl_sym_name(sym) ((char*)(((jl_sym_t*)(sym)) + 1))` (it will be unfortunate to lose the ability to look at `p/s name->name` in gdb though)